core/src/main/scala/cats/syntax/either.scala

```diff
@@ -124,51 +124,51 @@ final class EitherOps[A, B](val eab: Either[A, B]) extends AnyVal {
     case r @ Right(_) => EitherUtil.leftCast(r)
   }
 
-  def flatMap[D](f: B => Either[A, D]): Either[A, D] = eab match {
+  def flatMap[AA >: A, D](f: B => Either[AA, D]): Either[AA, D] = eab match {
     case l @ Left(_) => EitherUtil.rightCast(l)
     case Right(b)    => f(b)
   }
 
-  def compare(that: Either[A, B])(implicit A: Order[A], B: Order[B]): Int = eab match {
+  def compare[AA >: A, BB >: B](that: Either[AA, BB])(implicit AA: Order[AA], BB: Order[BB]): Int = eab match {
     case Left(a1)  =>
       that match {
-        case Left(a2) => A.compare(a1, a2)
+        case Left(a2) => AA.compare(a1, a2)
         case Right(_) => -1
       }
     case Right(b1) =>
       that match {
         case Left(_)   => 1
-        case Right(b2) => B.compare(b1, b2)
+        case Right(b2) => BB.compare(b1, b2)
       }
   }
 
-  def partialCompare(that: Either[A, B])(implicit A: PartialOrder[A], B: PartialOrder[B]): Double = eab match {
+  def partialCompare[AA >: A, BB >: B](that: Either[AA, BB])(implicit AA: PartialOrder[AA], BB: PartialOrder[BB]): Double = eab match {
```

Comment:
this method is untested. Is it too much to ask to write a test of some kind during this PR?